src/TestInfrastructure/Framework/ConfigurableSolutionBinding.cs

```diff
@@ -8,59 +8,42 @@
 using Microsoft.VisualStudio.TestTools.UnitTesting;
 using SonarLint.VisualStudio.Integration.Persistence;
 using System;
-using System.Collections.Generic;
 
 namespace SonarLint.VisualStudio.Integration.UnitTests
 {
     internal class ConfigurableSolutionBinding : ISolutionBinding
     {
-        private int pendingFiles;
-        private int writeSolutionBindingRequests;
+        private int writtenFiles;
 
         #region ISolutionBinding
         BoundSonarQubeProject ISolutionBinding.ReadSolutionBinding()
         {
+            this.ReadSolutionBindingAction?.Invoke();
             return this.CurrentBinding;
         }
 
-        string ISolutionBinding.WriteSolutionBinding(ISourceControlledFileSystem sccFileSystem, BoundSonarQubeProject binding)
+        string ISolutionBinding.WriteSolutionBinding(BoundSonarQubeProject binding)
         {
```

Comment:
As mentioned, this test implementation has nothing to do with ISourceControlledFileSystem, so we just log the writes instead of writes and requests to write that are now the same.